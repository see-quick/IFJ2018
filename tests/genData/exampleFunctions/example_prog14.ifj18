# GCD(Greatest Common Divider) of Two Numbers using Recursion

def modulo(divider, divident)
    result = divider - ((divider/divident)*divident)
end

def hcf(n1, n2)

  if n2 != 0 then
    partial = modulo(n1, n2)
    result = hcf(n2, partial)
  else
    result = n1
  end
end

print "Enter two positive integers: \n"
n1 = inputi
print "\n"
n2 = inputi
print "\n"

result = hcf(n1, n2)
print "GCD of ", n1, " and ", n2 , " is ", result
