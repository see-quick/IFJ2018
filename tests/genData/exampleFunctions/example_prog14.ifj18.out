Enter two positive integers: 
4
6
GCD of 4 and 6 is 2

